src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.MultiAgent.cs

```diff
@@ -559,14 +559,11 @@ private static size_t CurlReceiveHeadersCallback(IntPtr buffer, size_t size, siz
 
                         if (!TryParseStatusLine(response, responseHeader, easy))
                         {
-                            int colonIndex = responseHeader.IndexOf(':');
-
-                            // Skip malformed header lines that are missing the colon character.
-                            if (colonIndex > 0)
+                            int index = 0;
+                            string headerName = CurlResponseParseUtils.ReadHeaderName(responseHeader, out index);
+                            if (headerName != null)
                             {
-                                string headerName = responseHeader.Substring(0, colonIndex);
-                                string headerValue = responseHeader.SubstringTrim(colonIndex + 1);
-
+                                string headerValue = responseHeader.Substring(index + 1).Trim();
```

Comment:
Sorry, I assumed that Substring(index) would fail if out-of-bounds. Since that is handling it, the code would still work. Thanks for adding the test case